### DISCUSSION TITLE
Wrong legend value?

### DISCUSSION THREAD

**Discussion by bengtner** [RomRider/apexcharts-card#574] [Q&A]
I have chart showing energy prices. It seems like the legend value shows the wrong value. Se enclosed graph, where legend value show 1.4 SEK, but the graph indicates 1.5. The underlaying value is 1.495 when I grabbed the picture, so 1.5 is definitely more correct than 1.4.

![image](https://github.com/RomRider/apexcharts-card/assets/18145417/742d704e-d686-42db-bc62-8b4f4e91e847)

Is this a rounding issue? Or does it  picks the wrong value from the data series? 

The yaml code looks like this:

```
type: custom:apexcharts-card
graph_span: 24h
header:
  title: Dagens pris (kr/kWh)
  show: true
span:
  start: day
now:
  show: true
  label: Now
series:
  - entity: sensor.nordpool_kwh_se4_sek_3_10_0
    type: column
    name: Timpris
    stroke_width: 5
    data_generator: |
      return entity.attributes.raw_today.map((start, index) => {
        return [new Date(start["start"]).getTime(), entity.attributes.raw_today[index]["value"]];
      });
  - entity: sensor.nordpool_kwh_se4_sek_3_10_0
    attribute: mean
    stroke_width: 1
    name: Medelpris
```
Edit: It seems like the legend value does not change at all during the day. Maybe this due to the fact I do not plot the value of the sensor as such, but an attribute of the sensor? 24h later (once a new set of attributes is availble) the legend value do change to a different value. But I can't figure out what this value comes from. It is not the sensor value at midnight when the data is updated, not a peak value or the value by the end of the day. I will suppress this value since it is misleading.

**Comment by BobC76** [comment]
Hi see this thread for a workaround, I think the legend state may not be correct for generated data but there is a fix to have the current value in the header:

https://github.com/RomRider/apexcharts-card/issues/482
